src/manifolds/ProjectiveSpace.jl

```diff
@@ -0,0 +1,392 @@
+"""
+function manifold_dimension(M::AbstractProjectiveSpace{𝔽}) where {𝔽}
+    return manifold_dimension(get_embedding(M)) - real_dimension(𝔽)
+end
+
+"""
+    mean(
+        M::AbstractProjectiveSpace,
+        x::AbstractVector,
+        [w::AbstractWeights,]
+        method = GeodesicInterpolationWithinRadius(π/4);
+        kwargs...,
+    )
+
+Compute the Riemannian [`mean`](@ref mean(M::Manifold, args...)) of points in vector `x`
+using [`GeodesicInterpolationWithinRadius`](@ref).
+"""
+mean(::AbstractProjectiveSpace, ::Any...)
+
+function Statistics.mean!(
+    M::AbstractProjectiveSpace,
+    p,
+    x::AbstractVector,
+    w::AbstractVector;
+    kwargs...,
+)
+    return mean!(M, p, x, w, GeodesicInterpolationWithinRadius(π / 4); kwargs...)
+end
+
+function mid_point!(M::ProjectiveSpace, q, p1, p2)
+    z = dot(p1, p2)
+    cosθ = abs(z)
+    signz = sign_from_abs(z, cosθ)
+    q .= p1 .+ p2 .* signz'
+    project!(M, q, q)
+    return q
+end
+
+"""
+    normal_tvector_distribution(M::ProjectiveSpace{n,ℝ}, p, σ)
+
+Generate a distribution in the tangent space at `p` by generating a
+normal distribution in ambient space with standard deviation `σ`
+projected to the tangent space at `p`.
+"""
+function normal_tvector_distribution(M::ProjectiveSpace{n,ℝ}, p, σ) where {n}
+    d = Distributions.MvNormal(zero(p), σ)
+    return ProjectedFVectorDistribution(TangentBundleFibers(M), p, d, project!, p)
+end
+
+project!(::AbstractProjectiveSpace, q, p) = (q .= p ./ norm(p))
+
+project!(::AbstractProjectiveSpace, Y, p, X) = (Y .= X .- p .* dot(p, X))
+
+@doc raw"""
+    representation_size(M::AbstractProjectiveSpace)
+
+Return the size points on the [`AbstractProjectiveSpace`](@ref) `M` are represented as,
+i.e., the representation size of the embedding.
+"""
+@generated representation_size(::ArrayProjectiveSpace{N}) where {N} = size_to_tuple(N)
+@generated representation_size(::ProjectiveSpace{N}) where {N} = (N + 1,)
+
+function retract!(
+    M::AbstractProjectiveSpace,
+    q,
+    p,
+    X,
+    ::Union{ProjectionRetraction,PolarRetraction,QRRetraction},
+)
+    q .= p .+ X
+    return project!(M, q, q)
+end
+
+function Base.show(io::IO, ::ProjectiveSpace{n,𝔽}) where {n,𝔽}
+    return print(io, "ProjectiveSpace($(n), $(𝔽))")
+end
+function Base.show(io::IO, ::ArrayProjectiveSpace{N,𝔽}) where {N,𝔽}
+    return print(io, "ArrayProjectiveSpace($(join(N.parameters, ", ")); field = $(𝔽))")
+end
+
+"""
+    uniform_distribution(M::ProjectiveSpace{n,ℝ}, p) where {n}
+
+Uniform distribution on given [`ProjectiveSpace`](@ref) `M`. Generated points will be of
+similar type as `p`.
+"""
+function uniform_distribution(M::ProjectiveSpace{n,ℝ}, p) where {n}
+    d = Distributions.MvNormal(zero(p), 1.0)
+    return ProjectedPointDistribution(M, d, project!, p)
+end
+
+@doc raw"""
+    vector_transport_to(M::AbstractProjectiveSpace, p, X, q, method::ParallelTransport)
+
+Parallel transport a vector `X` from the tangent space at a point `p` on the
```

Comment:
A formula (maybe one that already holds for the Abstract one in general) would be nice,
Otherwise (since this should default back to direction?) There is no additional information in this string compared to the `Manifold` case?